lib/error_tracker/schemas/occurrence.ex

```diff
@@ -17,4 +20,54 @@ defmodule ErrorTracker.Occurrence do
 
     timestamps(type: :utc_datetime_usec, updated_at: false)
   end
+
+  @doc false
+  def changeset(occurrence, attrs) do
+    occurrence
+    |> cast(attrs, [:context, :reason])
+    |> maybe_put_stacktrace()
+    |> validate_required([:reason, :stacktrace])
+    |> validate_context()
+    |> foreign_key_constraint(:error)
+  end
+
+  # This function validates if the context can be serialized to JSON before
+  # storing it to the DB.
+  #
+  # If it cannot be serialized a warning log message is emitted and an error
+  # is stored in the context.
+  #
+  defp validate_context(changeset) do
+    if changeset.valid? do
+      context = get_field(changeset, :context, %{})
+
+      json_encoder =
+        ErrorTracker.Repo.with_adapter(fn
+          :postgres -> Application.get_env(:postgrex, :json_library, Jason)
+          :sqlite -> Application.get_env(:ecto_sqlite3, :json_library, Jason)
+        end)
```

Comment:
This is hacky, but it is the best solution I could find to avoid just hardcoding Jason.